libraries/core_libs/consensus/src/dag/dag.cpp

```diff
@@ -452,6 +453,35 @@ void DagManager::addDagBlock(DagBlock const &blk, SharedTransactions &&trxs, boo
                << " pivot = " << frontier_.pivot << " tips: " << frontier_.tips;
 }
 
+void DagManager::sendNonFinalizedBlocks(std::unordered_set<blk_hash_t> &&blocks_hashes,
+                                        const std::shared_ptr<network::tarcap::TaraxaPeer> &peer,
+                                        uint64_t peer_period) {
+  // Sending of blocks is in DagManager class to use DagManager mutex_ which ensures that there is no race condition
+  // between syncing and gossiping dag blocks
+  SharedLock lock(mutex_);
+  std::vector<std::shared_ptr<DagBlock>> dag_blocks;
+  // There is no point in sending blocks if periods do not match
+  if (peer_period == period_) {
+    peer->syncing_ = false;
+    for (auto &level_blocks : non_finalized_blks_) {
+      for (auto &block : level_blocks.second) {
+        const auto hash = block;
```

Comment:
I think that could remove line 468, if do "for (auto &hash : level_blocks.second)"

Reply:
const auto &hash

Reply:
Fixed